lib/msal-common/src/request/ScopeSet.ts

```diff
@@ -21,29 +21,49 @@ export class ScopeSet {
     // Boolean denoting whether scopes are required. Usually used for validation.
     private scopesRequired: boolean;
 
-    constructor(inputScopes: Array<string>, appClientId: string, scopesRequired: boolean) {
+    constructor(
+        inputScopes: Array<string>,
+        appClientId: string,
+        scopesRequired: boolean,
+        nonSPA?: boolean
+    ) {
         this.clientId = appClientId;
         this.scopesRequired = scopesRequired;
+
         // Filter empty string and null/undefined array items
-        const filteredInput = inputScopes ? StringUtils.removeEmptyStringsFromArray(inputScopes) : inputScopes;
+        const filteredInput = inputScopes
+            ? StringUtils.removeEmptyStringsFromArray(inputScopes)
+            : inputScopes;
+
         // Validate and filter scopes (validate function throws if validation fails)
         this.validateInputScopes(filteredInput);
-        const scopeArr = filteredInput ? StringUtils.trimAndConvertArrayEntriesToLowerCase([...filteredInput]) : [];
+
+        const scopeArr = filteredInput
+            ? StringUtils.trimAndConvertArrayEntriesToLowerCase([
+                ...filteredInput
+            ])
+            : [];
         this.scopes = new Set<string>(scopeArr);
         if (!this.scopesRequired) {
             this.appendScope(this.clientId);
         }
         this.originalScopes = new Set<string>(this.scopes);
-        this.replaceDefaultScopes();
+
+        // add default scopes
+        nonSPA ? this.replaceDefaultScopes() : this.addDefaultScopes();
```

Comment:
Why do we have two separate APIs essentially doing the same thing?

Reply:
so for SPA you remove the clientId and replace the scopes, for other flows, clientId as a scope is  not supported.  Raised #1390 to address this.